Tests/DatadogBenchmarkTests/DataCollection/RUMBenchmarkTests.swift

```diff
@@ -0,0 +1,33 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Datadog
+import XCTest
+
+class RUMBenchmarkTests: BenchmarkTests {
+    func testCreatingOneRUMEvent() {
+        let viewController = UIViewController()
+        Global.rum.startView(viewController: viewController)
+
+        measure {
+            Global.rum.addUserAction(type: .tap, name: "tap")
+        }
```

Comment:
I agree, the goal of measuring is to pinpoint the cost of an individual call or logical group of call.
Now, the way measuring works on android is that the `measure` block will actually be ran 50+ times, to compute a statistically relevant measure (average, min and max duration of the measured block). Maybe you could do something similar to get a more accurate result.

Reply:
let's say my operation takes 1ns in an ideal environment but when i test it, it takes 100ns due to other factors (1ns signal, 100ns noise).
if i run my operation 100 times, my signal-to-noise ratio becomes better (~100ns signal, ~100ns noise) so that i'll be more likely to catch it when my operation becomes slower one day.

but i haven't run those tests in my local, maybe it gives meaningful results already. that's why i'm asking

Reply:
> let's say my operation takes 1ns in an ideal environment but when i test it, it takes 100ns due to other factors (1ns signal, 100ns noise).
> if i run my operation 100 times, my signal-to-noise ratio becomes better (~100ns signal, ~100ns noise) so that it'll be more likely to catch it when my operation becomes slower one day.

This is what `XCTest's` [`measure(_:)` does](https://developer.apple.com/documentation/xctest/xctestcase/1496290-measure). It runs the measured code 10 times, by making sure it's well isolated (there are also [more advanced variants](https://developer.apple.com/documentation/xctest/xctestcase/1496277-measuremetrics)). In result you get the AVG time and max STDDEV describing the "noise":

<img width="276" alt="Screenshot 2020-11-18 at 12 49 26" src="https://user-images.githubusercontent.com/2358722/99527350-f19a7280-299c-11eb-951b-985f83a67fff.png">

What we do further in CI App is inferring more data, including `min/max`, `median`, `p99` etc.

Reply:
For now we keep the baselines not set, as we don't run benchmarks on physical device as part of our automation. But I see huge potential in trend analysis that we can do in CI App. Even if CI measurements significantly vary from one build to another, when smoothed by aggregation they would be describing meaningful trends.